web/core/components/issues/issue-layouts/properties/labels.tsx

```diff
@@ -49,318 +45,157 @@ export const IssuePropertyLabels: React.FC<IIssuePropertyLabels> = observer((pro
   if (storeLabels && storeLabels.length > 0) projectLabels = storeLabels;
 
-  const options = projectLabels.map((label) => ({
-    value: label?.id,
-    query: label?.name,
-    content: (
-      <div className="flex items-center justify-start gap-2 overflow-hidden">
-        <span
-          className="h-2.5 w-2.5 flex-shrink-0 rounded-full"
-          style={{
-            backgroundColor: label?.color,
-          }}
-        />
-        <div className="line-clamp-1 inline-block truncate">{label?.name}</div>
-      </div>
-    ),
-  }));
-
-  const filteredOptions =
-    query === "" ? options : options?.filter((option) => option.query.toLowerCase().includes(query.toLowerCase()));
-
-  const label = (
-    <div className="flex h-full w-full flex-wrap items-center gap-2 overflow-hidden">
+  return (
+    <>
       {value.length > 0 ? (
         value.length <= maxRender ? (
           <>
             {projectLabels
               ?.filter((l) => value.includes(l?.id))
               .map((label) => (
-                <Tooltip
+                <LabelDropdown
                   key={label.id}
+                  projectId={projectId}
+                  value={value}
+                  onChange={onChange}
+                  buttonClassName={buttonClassName}
+                  placement={placement}
+                  hideDropdownArrow={hideDropdownArrow}
+                  fullWidth={fullWidth}
+                  fullHeight={fullHeight}
+                  label={
+                    <Tooltip
+                      key={label.id}
+                      position="top"
+                      tooltipHeading="Labels"
+                      tooltipContent={label?.name ?? ""}
+                      isMobile={isMobile}
+                      renderByDefault={renderByDefault}
+                    >
+                      <div
+                        key={label?.id}
+                        className={`flex overflow-hidden justify-center hover:bg-custom-background-80 ${
+                          !disabled && "cursor-pointer"
+                        } h-full ${fullWidth && "w-full"} max-w-full flex-shrink-0 items-center rounded px-2.5 text-xs ${noLabelBorder ? "rounded-none" : "border-[0.5px] border-custom-border-300"}`}
+                      >
+                        <div className="flex max-w-full items-center gap-1.5 overflow-hidden text-custom-text-200">
+                          <span
+                            className="h-2 w-2 flex-shrink-0 rounded-full"
+                            style={{
+                              backgroundColor: label?.color ?? "#000000",
+                            }}
+                          />
+                          <div className="line-clamp-1 inline-block w-auto max-w-[200px] truncate">{label?.name}</div>
+                        </div>
+                      </div>
+                    </Tooltip>
+                  }
+                />
+              ))}
+          </>
+        ) : (
```

Comment:
These conditions are a bit harder to read. Is it possible to refactor it a bit better?

Reply:
Done.